File: Lab04/dice.py
import random


class Dice:
    """
    A class to represent a die.

    Attributes:
        __num_sides (int): The number of sides on the dice.
        __current_value (int): The current value showing on the dice.

    Methods:
        roll(): Rolls the dice to produce a new value.
        get_value(): Returns the current value of the dice.
    """

    def __init__(self, num_sides=6):
        """
        Constructs all the necessary attributes for the dice object.

        Parameters:
            num_sides (int): The number of sides on the dice, default is 6.
        """
        self.__num_sides = num_sides
        self.__current_value = self.roll()

    def roll(self):
        """
        Rolls the dice to generate a new value between 1 and the number of sides.

        Returns:
            int: The new value of the dice.
        """
        self.__current_value = random.randint(1, self.__num_sides)
        return self.__current_value

    def get_value(self):
        """
        Gets the current value showing on the dice.

        Returns:
            int: The current value of the dice.
        """
        return self.__current_value

File: Lab05/src/Dice.java
import java.util.Random;

/**
 * This class models a die with a specified number of sides.
 */
public class Dice {

    /**
     * The number of sides the dice has.
     */
    private int numSides;

    /**
     * The current value showing on the dice.
     */
    private int currentValue;

    /**
     * Constructor that takes the number of sides the dice should have.
     * Initializes the number of sides and sets the current value to 1.
     *
     * @param numSides The number of sides for the dice.
     */
    public Dice(int numSides) {
        this.numSides = numSides;
        this.currentValue = 1;
    }

    /**
     * Default constructor that creates a D6 (6-sided dice).
     */
    public Dice() {
        this(6);
    }

    /**
     * Rolls the dice (sets the current value to a new random number from 1 to the number of sides).
     */
    public void roll() {
        Random random = new Random();
        currentValue = random.nextInt(numSides) + 1;
    }

    /**
     * Returns the current value showing on the dice.
     *
     * @return The current value (1 to number of sides).
     */
    public int getValue() {
        return currentValue;
    }
}
